pkg/gatewayserver/io/basicstationlns/messages/downstream.go

```diff
@@ -0,0 +1,83 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package messages
+
+import (
+	"encoding/json"
+	"time"
+
+	"go.thethings.network/lorawan-stack/pkg/basicstation"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+)
+
+var errDownlinkMessage = errors.Define("downlink_message", "could not translate downlink message")
```

Comment:
So is this a translate error?

Reply:
I think so. In UDP this (conversion of packets to NS format) is called as `translation` so I stuck to that naming.